test/python/transpiler/test_light_cone.py

```diff
@@ -0,0 +1,294 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the LightCone pass"""
+
+import unittest
+from qiskit.circuit.library.n_local.efficient_su2 import efficient_su2
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+import numpy as np
+
+from qiskit.circuit import ParameterVector, QuantumCircuit, QuantumRegister
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.circuit.library import real_amplitudes
+from qiskit.quantum_info import Operator, Pauli
+from qiskit.transpiler.passes import RemoveFinalMeasurements
+from qiskit.transpiler.passes.optimization.light_cone import LightCone
+from qiskit.transpiler.passmanager import PassManager
+from qiskit.quantum_info import Statevector
+
+# Missing cases:
+#   - Rotational gates with notable angles (0, np.pi);
+#   - Light-cone with parametrized gates, see below and issue #12790.
+
+
+class TestLightConePass(QiskitTestCase):
+    """Test the LightCone pass."""
+
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+
+    def test_large_circuit(self):
+        """Test the LightCone pass can be run for large circuits."""
+
+        num_qubits = 120
+        qc = real_amplitudes(num_qubits, entanglement="pairwise", reps=3).decompose()
+        pm = PassManager([LightCone(Pauli("Z" + "I" * (num_qubits - 1)))])
+        pm.run(qc)
+
+    def test_parametrized(self):
+        """Test the LightCone pass with a parameterized circuit."""
+
+        num_qubits = 4
+        qc = real_amplitudes(num_qubits, entanglement="pairwise", reps=3)
+        pm = PassManager([LightCone(Pauli("Z" + "I" * (num_qubits - 1)))])
+        new_circuit = pm.run(qc)
+
+        self.assertEqual(new_circuit.num_parameters, 11)
+
+        qr = QuantumRegister(num_qubits)
+        expected = QuantumCircuit(qr)
+        theta = ParameterVector("θ", num_qubits**2)
+
+        expected.ry(theta[0], qr[0])
```

Comment:
Would it be simpler (and less error prone to type) to choose more specific, smaller (i.e. shallower/less ops) circuits to test all edge cases? 🙂